### PR TITLE
Adding write_file, write_string and allow for kwargs in initialization of Wasbhook

### PR DESCRIPTION
**Pull request by marcjimz**
This PR addresses the following changes:

1. Write a file from path to Azure Blob storage (WASBS)
2. Write a file from text to Azure Blob storage (WASBS)
3. Allow for kwargs to be passed into the WasbHook constructor

We need the ability to write back to Blob storage and this covers a missing gap in this hook.

Regarding the kwargs, we require the AirFlow pipelines to be generalizable at runtime, and this cannot be done without an existing connection made that authorizes the connection. This fix provides an alternative way to provide access. We prefer these connections to be ephemeral in nature, and not stored for longtime, and we can do this when executing a workload.

It should also be noted that these credentials should not be stored in the DAG itself but passed in as a property, such as a SAS Token.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by marcjimz** [comment]
I have made the equivalent changes for Airflow 2.0 and it is available here: https://github.com/marcjimz/airflow/blob/master/airflow/providers/microsoft/azure/hooks/wasb.py

**Comment by marcjimz** [comment]
Hi @mik-laj , thoughts on this one?

**Comment by RosterIn** [comment]
@marcjimz You need to raise PR against Airflow Master.
The changes you will make on Master will be available to 1.10.x releases with the providers packages.

**Comment by turbaszek** [comment]
> @marcjimz You need to raise PR against Airflow Master.

That's true, please rebase and change the target branch.

File: airflow/contrib/hooks/wasb_hook.py [hunk 1]
```diff
@@ -36,9 +36,17 @@ class WasbHook(BaseHook):
...
     """
 
-    def __init__(self, wasb_conn_id='wasb_default'):
+    def __init__(self, wasb_conn_id='wasb_default', **kwargs):
         self.conn_id = wasb_conn_id
-        self.connection = self.get_conn()
+        if 'sas_token' in kwargs and kwargs['sas_token'] is not None:
+            self.connection = self.get_custom_conn(**kwargs)
+        else:
+            self.connection = self.get_conn()
```
Comment (ashb) [inline airflow/contrib/hooks/wasb_hook.py:44]: Why do you need this when there's already a mechanism for passing kwargs to the BlockBlobService.

What is special about `sas_token` -- and if there are other values passed here they are just sliently ignored if that is false which doesn't seem very good behaviour.

I think a better approach would be to remove this if/else, to pass all kwargs to the existing `get_conn()` and have that mergbe the kwargs with the extra_dejson from the connection.

File: airflow/contrib/hooks/wasb_hook.py [hunk 2]
```diff
@@ -189,3 +197,50 @@ def delete_file(self, container_name, blob_name, is_prefix=False,
...
+        :param text: text content to be written to blob storage
+        :type text: str
+        :param encoding: Python encoding to use to convert the text to bytes
+        :type encoding: str
+        :param kwargs: Optional keyword arguments that
+            `BlockBlobService.create_blob_from_text()` takes.
+        :type kwargs: object
+        """
+        if text is None:
+            raise AirflowException("is_path variable set to FALSE but no text was provided.")
```
Comment (ashb) [inline airflow/contrib/hooks/wasb_hook.py:219]: What is_path?

**Review by ashb — CHANGES_REQUESTED** [review]
Please add unite tests for the new methods.

**Pull request closed without merging**